packages/aws-amplify/src/initSingleton.ts

```diff
@@ -36,6 +38,17 @@ export const DefaultAmplify = {
 		libraryOptions?: LibraryOptions,
 	): void {
 		const resolvedResourceConfig = parseAmplifyConfig(resourceConfig);
+		const cookieBasedKeyValueStorage = new CookieStorage({ sameSite: 'lax' });
+		const resolvedKeyValueStorage = libraryOptions?.ssr
+			? cookieBasedKeyValueStorage
+			: defaultStorage;
+		const cookieBasedCredentialsAndIdentityIdProvider =
+			new CognitoAWSCredentialsAndIdentityIdProvider(
+				new DefaultIdentityIdStore(cookieBasedKeyValueStorage),
+			);
+		const resolvedCredentialsProvider = libraryOptions?.ssr
+			? cookieBasedCredentialsAndIdentityIdProvider
+			: cognitoCredentialsProvider;
```

Comment:
Nit: an unknowing contributor could mistakenly provide `cookieBasedCredentialsAndIdentityIdProvider` to client lib options
```suggestion
const resolvedCredentialsProvider = libraryOptions?.ssr
			? new CognitoAWSCredentialsAndIdentityIdProvider(
					new DefaultIdentityIdStore(cookieBasedKeyValueStorage),
				)
			: cognitoCredentialsProvider;
```

Reply:
Good catch, updated.